src/calendar-list/new.tsx

```diff
@@ -0,0 +1,196 @@
+  }, [currentMonth]);
+
+  const getMonthIndex = useCallback((month?: XDate) => {
+    if (!month) {
+      return -1;
+    }
+    return items.findIndex(item => item.includes(month.toString('yyyy-MM')));
+  }, [items]);
+
+  const scrollToMonth = useCallback((month?: string) => {
+    if (month) {
+      const index = getMonthIndex(new XDate(month));
+      if (index !== -1) {
+        // @ts-expect-error
+        list.current?.scrollToOffset?.(index * constants.screenWidth, 0, true);
+      }
+    }
+  }, [getMonthIndex]);
+
+  const updateMonth = useCallback((count: number, month?: XDate) => {
+    if (month) {
+      const next = new XDate(month).addMonths(count, true);
+      const nextNext = new XDate(month).addMonths(count * 2, true);
+      const nextNextIndex = getMonthIndex(nextNext);
+      if (nextNextIndex !== -1) {
+        setCurrentMonth(toMarkingFormat(next));
+      }
+    }
+  }, [getMonthIndex]);
+
+  const scrollToNextMonth = useCallback((method: () => void, month?: XDate) => {
+    if (calendarProps?.onPressArrowLeft) {
+      calendarProps?.onPressArrowLeft?.(method, month);
+    } else {
+      updateMonth(1, month);
+    }
+  }, [updateMonth]);
+
+  const scrollToPreviousMonth = useCallback((method: () => void, month?: XDate) => {
+    if (calendarProps?.onPressArrowRight) {
+      calendarProps?.onPressArrowRight?.(method, month);
+    } else {
+      updateMonth(-1, month);
+    }
+  }, [updateMonth]);
+
+  const reloadPages = useCallback(
+    pageIndex => {
+      const newItems = getDatesArray(items[pageIndex], scrollRange);
+      setItems(newItems);
+    },
+    [items]
+  );
+
+  const onPageChange = useCallback((pageIndex: number, _: number, info: {scrolledByUser: boolean}) => {
+    if (useStaticHeader && info.scrolledByUser) {
+      setCurrentMonth(items[pageIndex]);
+    }
+  }, [items]);
+
+  const renderStaticHeader = () => {
+    if (useStaticHeader) {
+      return (
+        <CalendarHeader
+          {...headerProps}
+          month={new XDate(currentMonth)}
+          onPressArrowRight={scrollToNextMonth}
+          onPressArrowLeft={scrollToPreviousMonth}
+          style={staticHeaderStyle}
+          accessibilityElementsHidden // iOS
+          importantForAccessibility={'no-hide-descendants'} // Android
+          testID={'static-header'}
+        />
+      );
+    }
+  };
+
+  const renderItem = useCallback((_type: any, item: string) => {
+    return (
+      <Calendar
+        {...calendarProps}
+        {...headerProps}
+        initialDate={item}
+        disableMonthChange
+        onPressArrowRight={scrollToNextMonth}
+        onPressArrowLeft={scrollToPreviousMonth} 
+        hideExtraDays={calendarProps?.hideExtraDays || true}
+        style={[style.current.calendar, calendarProps?.style]}
+        headerStyle={horizontal ? calendarProps?.headerStyle : undefined}
+        testID={`${testID}_${item}`}
+        // context={context}
+      />
+    );
+  }, [calendarProps, items]);
```

Comment:
Technically shouldn't you depend on `scrollToNextMonth` and `scrollToPreviousMonth` that transitively depends on items. 
I think it's easier to understand the code when you see a function depends on something that it uses

Reply:
Correct. Fixed